minari/minari_storage.py

```diff
@@ -0,0 +1,119 @@
+import os
+from typing import Any, Callable, Iterable, List, Optional, Union
+import gymnasium as gym
+import h5py
+from gymnasium.envs.registration import EnvSpec
+
+
+_PathLike = Union[str, bytes, os.PathLike]
+
+
+class MinariStorage:
+    def __init__(self, data_path: _PathLike):
+        """Initialize properties of the Minari storage.
+
+        Args:
+            data_path (str): full path to the `main_data.hdf5` file of the dataset.
+        """
+        self._data_path = data_path
+        self._extra_data_id = 0
+        with h5py.File(self._data_path, "r") as f:
+            self._flatten_observations = f.attrs["flatten_observation"]
+            self._flatten_actions = f.attrs["flatten_action"]
```

Comment:
The user should have access to these attributes through `MinariDataset`. I suggest having a `MinaraDatasetSpec` attribute that contains all of these attributes and that is accessible through `MinariDataset`. We can do this in a later PR